kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicaSetOperationsImpl.java

```diff
@@ -133,13 +133,18 @@ private void waitUntilScaled() {
 
     final Runnable rcPoller = new Runnable() {
       public void run() {
-        ReplicaSet rc = getMandatory();
-        atomicRC.set(rc);
-        if (rc.getStatus().getObservedGeneration() >= rc.getMetadata().getGeneration() && Objects.equals(rc.getSpec().getReplicas(), rc.getStatus().getReplicas())) {
-          countDownLatch.countDown();
-        } else {
+        try {
+          ReplicaSet rc = getMandatory();
+          atomicRC.set(rc);
+          if (rc.getStatus() == null || rc.getStatus().getObservedGeneration() == null || rc.getMetadata().getGeneration() == null) {
```

Comment:
Bad formatting, plus shouldn't have an empty `if` block. Just add it to the conditional below.